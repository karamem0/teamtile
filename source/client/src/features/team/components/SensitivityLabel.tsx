//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import Presenter from './SensitivityLabel.presenter';

interface SensitivityLabelProps {
  value?: string
}

function SensitivityLabel(props: SensitivityLabelProps) {

  const { value } = props;

  return (
    <Presenter value={value} />
  );

}

export default SensitivityLabel;
