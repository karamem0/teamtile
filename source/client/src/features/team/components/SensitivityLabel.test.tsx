//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import SensitivityLabel from './SensitivityLabel.presenter';
import { render } from '@testing-library/react';

test('create shapshot of when SensitivityLabel is not null', async () => {
  const params = {
    value: 'Restricted'
  };
  const { asFragment } = render(<SensitivityLabel {...params} />);
  expect(asFragment()).toMatchSnapshot();
});

test('create shapshot of when SensitivityLabel is undefined', async () => {
  const params = {
    value: undefined
  };
  const { asFragment } = render(<SensitivityLabel {...params} />);
  expect(asFragment()).toMatchSnapshot();
});
