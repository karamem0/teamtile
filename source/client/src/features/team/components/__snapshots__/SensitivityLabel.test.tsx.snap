// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot when the label is not undefined 1`] = `
<DocumentFragment>
  <div
    class="fui-Badge r1iycov ___172bkpj_b7iz950 f16xkysk faj9fo0 f161y7kd f1c8dzaj f1619yhw f1sl6hi9"
  >
    Restricted
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot when the label is undefined 1`] = `<DocumentFragment />`;
